provider/provider_test.go

```diff
@@ -157,3 +161,127 @@ func testProviderUpgrade(t *testing.T, example string) {
 	)
 	assertpreview.HasNoReplacements(t, result)
 }
+
+type testProviderCodeChangesOptions struct {
```

Comment:
This was adapted from here: https://github.com/pulumi/pulumi-aws/blob/2a3dd788d66627edd57ed38c86a6f761c65eea64/provider/provider_test.go#L121

I wanna move this to https://github.com/pulumi/providertest but first wanted to get this test working